wunderio/drupal-project
# QAG-49: Add DDEV support with integrated services

## Ticket QAG-49, GH-420

# Add DDEV support with integrated services

## Overview
This PR introduces DDEV as an alternative local development environment alongside Lando, providing a complete set of development services and improved documentation.

## Changes

### DDEV Integration
- Set up initial DDEV configuration matching Lando features
- Integrate essential services:
  - Adminer for database management
  - Elasticsearch 8.10.2
  - Varnish caching layer
- Automate composer install in DDEV startup
- Replace Kibana with Elasticvue for Elasticsearch UI
- Update Adminer addon to v1.1.0
- Configure Xdebug settings

### Documentation
- Add comprehensive DDEV environment documentation
- Improve local setup instructions with complete workflow
- Consolidate services documentation
- Enhance database sync documentation
- Improve README readability with collapsible sections

### Performance & Configuration
- Enable state cache for Drupal 10
- Add configuration sync directory
- Update Elasticsearch configuration to version 8.10.2

### Additional Features
- Add / update Cursor AI Code Editor documentation and rules

## Related Tickets
- GH-420: Initial DDEV implementation
- QAG-49: DDEV environment improvements
- QAG-57: Drupal performance optimization
- QAG-58: Cursor AI documentation
- QAG-60: Elasticsearch configuration updates

## Testing Steps
1. Clone the repository
2. Start DDEV environment: `ddev start`
3. Authenticate SSH for database syncing: `ddev auth ssh`
4. Verify all services are accessible:
   - Main site: https://drupal-project.ddev.site
   - Other services via `ddev describe`
5. Test database sync: `ddev syncdb`
6. Get a one-time login link for admin access `ddev drush uli`

## Commits

- 013cb84 QAG-58: (docs) Add Cursor AI Code Editor documentation
- 269df82 QAG-58: (docs) Emphasize default use of git diff --staged | cat for commit messages
- f2eaa58 QAG-49: (docs) Improve DDEV setup instructions with complete workflow
- dade727 QAG-49: (docs) Improve README readability with collapsible sections
- 229378b QAG-49: (feat) Automate composer install in DDEV startup
- cff837d QAG-49: (feat) Replace Kibana with Elasticvue for Elasticsearch UI
- 1c18974 QAG-60: (chore) Update Lando Elasticsearch configuration to version 8.10.2
- c7f01fd QAG-49: (docs) Improve DDEV services documentation with dynamic approach
- 646153f QAG-49: (chore) Update DDEV Adminer addon to v1.1.0
- 45ca66e QAG-57: (perf) Enable state cache for Drupal 10
- c732b97 QAG-49: (docs) Add explanation for DDEV settings management option
- 135a7b0 QAG-49: (chore) Disable Xdebug in DDEV configuration
- 0307cc3 QAG-60: (feat) Consolidate DDEV Elasticsearch 8 configuration
- c774fb1 QAG-49: (fix) Add missing configuration sync directory
- 41082ae QAG-49: (docs) Consolidate local environment setup instructions
- d056cde QAG-49: (docs) Consolidate services documentation in README
- c9a41d7 QAG-49: (fix) Fix Lando section collapsible formatting in README
- d25df1d QAG-49: (docs) Improve database sync documentation and implementation
- e63321f QAG-49: (docs) Fix bare URLs in Markdown files to comply with MD034 linting rule
- d20a49f QAG-49: (docs) Add DDEV environment documentation and improve local setup sections
- e1cafd1 GH-420: Initial README updates for DDEV setup
- 01ecba7 GH-420: Matching more DDEV config with Lando
- e57d748 GH-420: Varnish for DDEV
- c489643 GH-420: Kibana for DDEV
- 29deea6 GH-420: Elasticsearch for DDEV
- 431634f GH-420: Add Adminer for DDEV
- 3ac626f GH-420: Initial DDEV support

## Old changes description

Add a DDEV setup that works alongside Lando. Contains basic services like Adminer, Elasticsearch, GrumPHP, Kibana, Mailpit, Varnish, and some Lando commands converted to have DDEV support (GrumPHP, PHPUnit).

### Summary
- Update README with instructions on how to get DDEV environment up and running
- Adminer
  - Use default configuration by `ddev-adminer`
- Elasticsearch
  - Use default ES configuration by `ddev-elasticsearch`
  - Use `docker-compose.elasticsearch_extra.yaml` to only override the image to use ES8 + install `analysis-ukrainian` and `analysis-icu`.
- Kibana
  - Use default configuration by `ddev-kibana`
  - Use `docker-compose.kibana_extra.yaml` to only override the image version to match ES
  - Worth to note that [ddev-kibana](https://github.com/JanoPL/ddev-kibana/) seems to be quite small project, could be replaced with just a custom Dockerfile
- Mailpit
  - Replaces the age-old Mailhog. Supplied by DDEV.
- Varnish
  - Default `ddev-varnish` configuration
  - The small vcl bit from `.lando/varnish.vcl` added to `default.vcl`

## Some more things to do and consider

- [ ] Consider ignoring some of the default files that are coming from `ddev get`. For example the `ddev get ddev/ddev-varnish` adds a bunch of commands under `.ddev/commands/varnish` that are `#ddev-generated` and might not need to be in the repo
- [ ] Find out the best way to initially disable some services like we now have for Lando, like ES, Kibana and Varnish (~docker profiles (?)~ [`omit_containers`](https://ddev.readthedocs.io/en/stable/users/configuration/config/#omit_containers) might be the best way to do this)
- [x] How to keep the README more tidy as the double docs for both environments aren't very nice
- [x] Find out rest of the things that are now Lando-specific and convert them to DDEV equivalents
- [x] Probably some more things

## How to test

### Testing in feature environment:

No feature environment to test in.

### Local testing

Follow the instructions in the README.

### Testing steps

Confirm that existing Lando services work on DDEV.
